Client/lib/gMap.js

```diff
@@ -0,0 +1,231 @@
+      if (status == google.maps.DirectionsStatus.OK) {
+        directionsDisplay.setDirections(response);
+      }
+    });
+  }
+
+  function addMarker (latLng, title){
+    var id = markerCords.length;
+    markerCords.push(latLng);
+    var size = {
+      height: 40,
+      width: 40
+    }
+
+    var image = {
+      url: 'https://cdn2.iconfinder.com/data/icons/snipicons/500/map-marker-512.png',
+      scaledSize: new google.maps.Size(size.width,size.height),
+      origin: new google.maps.Point(0,0),
+      anchor: new google.maps.Point(size.width/2,size.height)
+    };
+
+    var marker = new google.maps.Marker({
+        position: latLng,
+        map: map,
+        title: title,
+        icon: image,
+    });
+
+    google.maps.event.addListener(marker, 'click', function() {
+      console.log(id);
+    });
+    if(!!lastLocationClicked){
+      calcRoute();
+    }
+    lastLocationClicked = latLng;
+  }
+
+  function CenterControl (controlDiv, text, callback, addToList,startAsActive) {
+
+    // Set CSS for the control border
+    var controlUI = document.createElement('div');
+    controlUI.className = 'gMapInactiveButton gMapButton';
+    if(startAsActive){
+      controlUI.className = 'gMapActiveButton gMapButton';
+    }
+    controlUI.style.border = '2px solid #fff';
+    controlUI.style.borderRadius = '200px';
+    controlUI.style.boxShadow = '0 2px 6px rgba(0,0,0,.3)';
+    controlUI.style.cursor = 'pointer';
+    controlUI.style.marginBottom = '22px';
+    controlUI.style.marginLeft = '10px';
+    controlUI.style.textAlign = 'center';
+    controlUI.style.width = '35px';
+    controlUI.style.height = '35px';
+    controlUI.style.marginTop = '10px';
+
+
+    controlUI.title = 'Click to recenter the map';
+
+    controlDiv.appendChild(controlUI);
+
+    // Set CSS for the control interior
+    var controlText = document.createElement('div');
+    controlText.className = 'gMapText';
+    controlText.style.color = 'rgb(255,255,255)';
+    controlText.style.fontFamily = 'Roboto,Arial,sans-serif';
+    controlText.style.fontSize = '16px';
+    controlText.style.lineHeight = '38px';
+    controlText.style.paddingLeft = '5px';
+    controlText.style.paddingRight = '5px';
+    controlText.style.paddingBottom = '5px';
+
+    controlText.innerHTML = text;
+    controlUI.appendChild(controlText);
+    if(addToList){
+      travelButtonList.push(controlUI);
+    }
+    // Setup the click event listeners: simply set the map to
+    // Chicago
+    google.maps.event.addDomListener(controlUI, 'click', function(){
+      if(addToList){
+        for (var i = 0; i < travelButtonList.length; i++) {
+          travelButtonList[i].className = 'gMapInactiveButton';
+          if(travelButtonList[i] === controlUI){
+            controlUI.className = 'gMapActiveButton';
+          }
+        };
+      }
+      callback();
+    });
+  }
+
+  function getGeolocation (callback){
+    var pos = new google.maps.LatLng(-33.73, 149.02);
```

Comment:
The map needs a lat and lng object to load, so we need a fallback/default location if all else fails that is what is location is, the fallback position. I believe a fallback location is needed for graceful failure.